Client develop

ayodelevm opened the pull request.

ayodelevm committed:
setting up thhe requirements and dependencies for react and redux

ayodelevm committed:
setting up webpack for react and redux

ayodelevm committed:
Adding dependencies to package.json file

ayodelevm committed:
updating eslint config files

ayodelevm committed:
ignoring bundled files

ayodelevm committed:
updating dependencies

ayodelevm committed:
updating webpack settings for reloading pages and loaders

ayodelevm committed:
adding html file

ayodelevm committed:
adding user actions file and file containing user action types

ayodelevm committed:
add functional components for signup and login page

ayodelevm committed:
add class component containing state

ayodelevm committed:
add reducers for signup and signin components

ayodelevm committed:
adding file connecting root reducer, and all middlewares to store

ayodelevm committed:
adding utility files containing methods for some http requests and file containving the different urls

ayodelevm committed:
adding file containing topmost component setting such as provider and routes and other imports

ayodelevm committed:
adding js files for materialize components

ayodelevm committed:
updating css styles for components

ayodelevm committed:
adding function for cross origin requests and changing status codes in auth ctrl

ayodelevm committed:
custom styling for new template design

ayodelevm committed:
new template design for post it with css and js files

ayodelevm committed:
ignoring ds_store file in templates

ayodelevm committed:
adding global variables to eslint file

ayodelevm committed:
refactoring landing, register and login pages to new template design

ayodelevm committed:
installing new depedencies

ayodelevm committed:
styling for new template design

ayodelevm committed:
redesigning templates for landing, register and signin pages

ayodelevm committed:
refactoring components

ayodelevm committed:
installig new depedencies

ayodelevm committed:
chore(validations|middlewares): adding new server-side validations and rewriting new middleware to authenticate user

ayodelevm committed:
chore(refactoring): removing all setup for passport authentication

ayodelevm committed:
chore(refactoring): allow for new columns to allow for new functionalities

ayodelevm committed:
chore(refactoring): rewrite signup and login functions using bcrypt and jwt and refactored routes accordingly

ayodelevm committed:
chore(refactoring): rewrite signup and login functions using bcrypt and jwt and refactored routes accordingly

ayodelevm committed:
moving middleware file to utils folder

ayodelevm committed:
Merge branch 'client-develop' into refactor-server

ayodelevm committed:
Merge pull request #22 from ayodelevm/refactor-server

Refactor server

ayodelevm committed:
Merge pull request #23 from ayodelevm/feature/client-components

refactoring components

ayodelevm committed:
chore(refactor): adding logout and set current user actions and refactoring actions reducers and container components

ayodelevm committed:
chore(refactor): adding logout and set current user actions and refactoring actions reducers and container components

ayodelevm committed:
chore(redesign templates): html, css and js files for template redesign for create group feature

ayodelevm committed:
chore(redesign templates): html, css and js files for template redesign for create group feature

ayodelevm committed:
Merge pull request #24 from ayodelevm/feature/client-components

chore(refactor): adding logout and set current user actions and refac…

ayodelevm committed:
Merge branch 'client-develop' into feature/redesign-templates

ayodelevm committed:
Merge pull request #25 from ayodelevm/feature/redesign-templates

chore(redesign templates): html, css and js files for template redesi…

ayodelevm committed:
chore(new chores): add action creators for addusers functions, group functions and messages function

ayodelevm committed:
chore(new chores): add action creators for addusers functions, group functions and messages function

ayodelevm committed:
chore(refactor) add redirect to dashboard function to signup and login form

ayodelevm committed:
chore(refactor) add redirect to dashboard function to signup and login form

ayodelevm committed:
chore(newChores): setup dashboard components

ayodelevm committed:
chore(newChores): setup dashboard components

ayodelevm committed:
(chore(newChore): added new routes for to put new task to  use

ayodelevm committed:
(chore(newChore): added new routes for to put new task to  use

ayodelevm committed:
chore(refactor): update utility file for api urls and endpoints

ayodelevm committed:
chore(refactor): update utility file for api urls and endpoints

ayodelevm committed:
chore(refactor): styling for dashboard

ayodelevm committed:
chore(refactor): styling for dashboard

ayodelevm committed:
chore(validations): validations for create group input

ayodelevm committed:
chore(validations): validations for create group input

ayodelevm committed:
chore(refactor): updating cors access for token authorization

ayodelevm committed:
chore(refactor): updating cors access for token authorization

ayodelevm committed:
chore(design): styling for dashboard and adding error color to create group form

ayodelevm committed:
chore(design): styling for dashboard and adding error color to create group form

ayodelevm committed:
chore(refactor): refactor validations for group control and convert key in error message from message to globals

ayodelevm committed:
chore(refactor): refactor validations for group control and convert key in error message from message to globals

ayodelevm committed:
chore(new): add validation for group input for client-side, add function for authorizing routes, add token to headers

ayodelevm committed:
chore(new): add validation for group input for client-side, add function for authorizing routes, add token to headers

ayodelevm committed:
chore(new): set current user in store

ayodelevm committed:
chore(new): set current user in store

ayodelevm committed:
feat(reducers): add reducers for group, mesage, auth and adduser functions and combine them in rootreducer

ayodelevm committed:
feat(reducers): add reducers for group, mesage, auth and adduser functions and combine them in rootreducer

ayodelevm committed:
feat(containers): add dashboard, group form, navbuttons, login form, signup form containers

ayodelevm committed:
feat(containers): add dashboard, group form, navbuttons, login form, signup form containers

ayodelevm committed:
feat(components): add dashboard, group modal, landing and nav components

ayodelevm committed:
feat(components): add dashboard, group modal, landing and nav components

ayodelevm committed:
feat(actions): add auth, group ad addusers actions

ayodelevm committed:
feat(actions): add auth, group ad addusers actions

ayodelevm committed:
chore(new): import new dependencies into entry point

ayodelevm committed:
chore(new): import new dependencies into entry point

ayodelevm committed:
chore(new): adding new globals to eslintrc file and new depedencies to package.json file

ayodelevm committed:
chore(new): adding new globals to eslintrc file and new depedencies to package.json file

ayodelevm committed:
removed material-ui initializations

ayodelevm committed:
removed material-ui initializations

ayodelevm committed:
remove console.log statements in actions

ayodelevm committed:
remove console.log statements in actions

ayodelevm committed:
chore(refactor): refactored components and added chips to create group forms and separate container components from presentational component, and converted inputs to reusable input

ayodelevm committed:
chore(refactor): refactored components and added chips to create group forms and separate container components from presentational component, and converted inputs to reusable input

ayodelevm committed:
chore(refactor): refactored inputs to reusable inputs and refactored container components

ayodelevm committed:
chore(refactor): refactored inputs to reusable inputs and refactored container components

ayodelevm committed:
removed console.log from apis

ayodelevm committed:
removed console.log from apis

ayodelevm committed:
refactored group Reducer

ayodelevm committed:
refactored group Reducer

ayodelevm committed:
refactored group controller to accept usernames of users to add to group when creating new group instead of user ids

ayodelevm committed:
refactored group controller to accept usernames of users to add to group when creating new group instead of user ids

ayodelevm committed:
removed some dependencies

ayodelevm committed:
removed some dependencies

ayodelevm committed:
feat(design): html, css and js files for template redesign

ayodelevm committed:
feat(design): html, css and js files for template redesign

ayodelevm committed:
Merge pull request #26 from ayodelevm/feature/client-components-groups

Feature/client components groups

ayodelevm committed:
Merge branch 'client-develop' into feature/redesign-templates

ayodelevm committed:
Merge pull request #27 from ayodelevm/feature/redesign-templates

feat(design): html, css and js files for template redesign

ayodelevm committed:
renamed navButtons to navContainer

ayodelevm committed:
renamed navButtons to navContainer

ayodelevm committed:
Merge pull request #28 from ayodelevm/client-components-dashboard

renamed navButtons to navContainer

ayodelevm committed:
refactoring html template file and installing new dependencies

ayodelevm committed:
refactoring html template file and installing new dependencies

ayodelevm committed:
refactoring controllers and models to accomodate changes in client-side

ayodelevm committed:
refactoring controllers and models to accomodate changes in client-side

ayodelevm committed:
changing redirect function and adding new route

ayodelevm committed:
changing redirect function and adding new route

ayodelevm committed:
changing from react-router-dom to react-router-redux and refactoring reducers

ayodelevm committed:
changing from react-router-dom to react-router-redux and refactoring reducers

ayodelevm committed:
refactoring container components and adding new functions to container components

ayodelevm committed:
refactoring container components and adding new functions to container components

ayodelevm committed:
refactored entry file to accomodate change in router and removing of commented code

ayodelevm committed:
refactored entry file to accomodate change in router and removing of commented code

ayodelevm committed:
Refactoring group and message actions

ayodelevm committed:
Refactoring group and message actions

ayodelevm committed:
adding more presentational components and adding more functionality

ayodelevm committed:
adding more presentational components and adding more functionality

ayodelevm committed:
Merge branch 'Review1' into client-develop

chukwumaijem commented on client/dev/js/actions/groupActions.js:
```diff
@@ -0,0 +1,105 @@
+  failure
+});
+
+// const token = window.localStorage.token;
```
Remove commented out code.

chukwumaijem commented on client/dev/js/components/Dashboard.js:
```diff
@@ -0,0 +1,72 @@
+};
+
+Dashboard.propTypes = {
+  // eslint-disable-next-line
```
Consider turning off rules you can't obey.

ayodelevm committed:
Merge pull request #29 from ayodelevm/feature/client-components-dashboard

Feature/client components dashboard

chukwumaijem commented on client/dev/js/components/Home.js:
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+
+require('../../../../scss/main.scss');
```
When paths start getting long, webpack is your friend. you can resolve paths so its easier to reference.

chukwumaijem commented on client/dev/js/actions/groupActions.js:
```diff
@@ -0,0 +1,105 @@
+  ).catch((err) => {
+    dispatch(createGroupFailure(err.message));
+  });
+};
```
Methods like this can quickly become hard to read. Consider abstracting the functions and make the then a one liner
`.then(handleSuccess, handleError)`

ayodelevm committed:
removed hammer.js from webpack and updated index.html file

ayodelevm committed:
removed hammer.js from webpack and updated index.html file

ayodelevm committed:
added babel-preset-stage-2

ayodelevm committed:
added babel-preset-stage-2

ayodelevm committed:
feat(actions): added new actions for dashboard and messaging container feature

ayodelevm committed:
feat(actions): added new actions for dashboard and messaging container feature

ayodelevm committed:
removed authoriczation from messaging container route

ayodelevm committed:
removed authoriczation from messaging container route

ayodelevm committed:
feat(components): modified conditions determining how groups and messages are rendered in the views

ayodelevm committed:
feat(components): modified conditions determining how groups and messages are rendered in the views

ayodelevm committed:
feat(containers):  modified and refactored container components and changed from passing props to presentational components through container state to passing directly from redux store to allow for re-render on new action

ayodelevm committed:
feat(containers):  modified and refactored container components and changed from passing props to presentational components through container state to passing directly from redux store to allow for re-render on new action

ayodelevm committed:
feat(reducers): modified and refactored reducers based on changes to actions

ayodelevm committed:
feat(reducers): modified and refactored reducers based on changes to actions

ayodelevm committed:
added ownerId to returned object after group has been created

ayodelevm committed:
added ownerId to returned object after group has been created

ayodelevm committed:
Merge pull request #31 from ayodelevm/feature/client-components-dashboard

Feature/client components dashboard

ayodelevm committed:
setting up webpack for production and serving react file from app.js file

ayodelevm committed:
setting up webpack for production and serving react file from app.js file

ayodelevm committed:
changing extensions from js to jsx

ayodelevm committed:
changing extensions from js to jsx

ayodelevm committed:
setting up app.js to send html file

ayodelevm committed:
setting up app.js to send html file

ayodelevm committed:
refactoring app file and url endpoints

ayodelevm committed:
refactoring app file and url endpoints

ayodelevm committed:
refactor for hosting on heroku

ayodelevm committed:
refactor for hosting on heroku

ayodelevm committed:
Delete package-lock.json

ayodelevm committed:
Delete package-lock.json

ayodelevm committed:
Adding loaders as dependencies

ayodelevm committed:
Adding loaders as dependencies

ayodelevm committed:
Merge pull request #32 from ayodelevm/webpack-heroku

Webpack heroku

ayodelevm committed:
feat(add|view users): reducers, actions, components and containers for adding users to group and viewing users

ayodelevm committed:
feat(add|view users): reducers, actions, components and containers for adding users to group and viewing users

ayodelevm committed:
refactoring addusers to group to allow for multiple users

ayodelevm committed:
refactoring addusers to group to allow for multiple users

ayodelevm committed:
Merge pull request #33 from ayodelevm/feature/client-components-users

Feature/client components users

ayodelevm committed:
feat(addUsers):  presentational and container components and reducer for adding users, implement filtering out group members from all users

ayodelevm committed:
feat(addUsers):  presentational and container components and reducer for adding users, implement filtering out group members from all users

ayodelevm committed:
refactoring add users to group ctrl and refactored npm scripts

ayodelevm committed:
refactoring add users to group ctrl and refactored npm scripts

ayodelevm committed:
Merge pull request #34 from ayodelevm/feature/client-components-users

Feature/client components users

ayodelevm committed:
removing firebase configuration

ayodelevm committed:
removing firebase configuration

ayodelevm committed:
feat(uploads): actions, reducers, containers and presentational components for file upload feature

ayodelevm committed:
feat(uploads): actions, reducers, containers and presentational components for file upload feature

ayodelevm committed:
refactor and move updateOneUser function from authCtrl and routes to addUserCtrl and routes

ayodelevm committed:
refactor and move updateOneUser function from authCtrl and routes to addUserCtrl and routes

ayodelevm committed:
remove firebase dependency

ayodelevm committed:
remove firebase dependency

ayodelevm committed:
Merge pull request #35 from ayodelevm/feature/upload-image

Feature/upload image

ayodelevm committed:
feat(templates): refactor templates for archiving and google sign-in functions

ayodelevm committed:
map scss to css

ayodelevm committed:
Merge branch 'client-develop' into feature/redesign-templates

ayodelevm committed:
Merge pull request #36 from ayodelevm/feature/redesign-templates

Feature/redesign templates

ayodelevm committed:
fix merge conflict

ayodelevm committed:
feat(archiving): actions for archiving and viewing archived messages

ayodelevm committed:
feat(archiving): presentational and container for archiving and viewing archived messages

ayodelevm committed:
feat(archiving): reducers and store for archiving  Messages in groups

ayodelevm committed:
feat(archiving): refactor  and write new server controllers and  routes to  allow for archiving messages and viewing arhived messages functionality

ayodelevm committed:
Merge pull request #37 from ayodelevm/feature/150580221/archive-messages

#150580221 Archive and View Archived Messages

ayodelevm committed:
installed nodemailer as new dependency

ayodelevm committed:
feat(reset-password): Add new route for reset password

ayodelevm committed:
feat(reset-password): Reset password actions and action types

ayodelevm committed:
feat(reset-password): presentational components for reset password feature

ayodelevm committed:
feat(reset-password): presentational components for reset password feature and other refactorinig

ayodelevm committed:
feat(reset-password): reducers for reset password feature and connect reducer to store

ayodelevm committed:
chore(reset-password): validations for reset passsword feature and refactor other validations

ayodelevm committed:
feat(reset-password): routes and controls for reset-password feature

ayodelevm committed:
moved scss folder from root into client folder and changed path in index file

ayodelevm committed:
feat(reset-password): reduce expiration time on link generated to reset password

ayodelevm committed:
Merge pull request #38 from ayodelevm/feature/150580005/reset-password

#150580005 Reset Password

ayodelevm committed:
chore(test): change test implementation from session to token, updated seed data and refactored status code

ayodelevm committed:
chore: refactored key in response object and response messages, and refactored status code

ayodelevm committed:
added timeout to coverage script

ayodelevm committed:
removing timeout from coverage script

ayodelevm committed:
Merge pull request #39 from ayodelevm/chore/refactor-server-tests

Chore/refactor server tests

ayodelevm committed:
Add jsdoc to functions and classes

ayodelevm committed:
Fix coverage failing issue

ayodelevm committed:
Merge pull request #42 from ayodelevm/js/docs

Js/docs

ayodelevm committed:
add socket.io server and client packages to dependecies

ayodelevm committed:
feat(sockets): move updating of messages with new message from message actions to socket actions, add implentaion for in-app notification and other real-time implementations

ayodelevm committed:
feat(sockets): initialize socket.io-client, move some functionalities from container component to sockets

ayodelevm committed:
feat(sockets): initialize socket.io for server-side

ayodelevm committed:
feat(sockets|notification): implement real-time in some features and initialize function for sms and email notification

ayodelevm committed:
feat(notification): implement email and sms notification based on urgent and critical priority levels

ayodelevm committed:
increase timeout for test

ayodelevm committed:
Merge pull request #43 from ayodelevm/feature/150580049/notifications

#150580049 Notifications

ayodelevm committed:
feat(google-signin): add google-auth-library and  react-google-login to depedencies

ayodelevm committed:
feat(google-signin): change parent componet for signup and signin

ayodelevm committed:
feat(google-signin): add actions for google signup and signin

ayodelevm committed:
feat(google-signin): add the presentational comnent for googe signin and signup and refactored input forms into a reusable component

ayodelevm committed:
feat(google-signin): rename signupform and loginform to signup container and login container, implement google signup and signin  functionality, move forms inputs into presentational component

ayodelevm committed:
feat(google-signin): add reducer for google-signup and signin feature

ayodelevm committed:
feat(google-signin): add google-signup and signin urls

ayodelevm committed:
feat(google-signin): add controller for google signup and signin and refactor validation

ayodelevm committed:
add new model for user and messages join-table and refactor user and message model

ayodelevm committed:
feat(google-signin): add google signup and google sign in routes

ayodelevm committed:
removed message sender from recipient of sms and email

ayodelevm committed:
Merge pull request #44 from ayodelevm/feature/150580009/google-signup-signin

Feature/150580009/google signup signin

ayodelevm committed:
chore(test): client side tests which includes test for:

- client side actions
- client side container and presentational components

ayodelevm committed:
fix(fix/feedback): Implement feedback and some fixes as listed below:

- Prevent access to app routes for unauthenticated users
- Prevent access to register, login and resetpassword pages for authenticated users
- Add 404 page not found component and route
- Add route to verify token on page reload for authenticated users
- rename addUsers file to users
- Add tooltip for icons
- refactore status code for ctrl and test files
- Adde JsDoc
- refactore namings of functions and objects
- move routes into a separate file

ayodelevm committed:
fix(fix/feedback): add tooltip to  icons

ayodelevm committed:
Merge pull request #45 from ayodelevm/chore/150790281/client-side-test

#150790281 client side tests

ayodelevm committed:
Merge pull request #46 from ayodelevm/fix/feedback-implementation

Fix/feedback implementation

ayodelevm committed:
chore(tests): implemeted client side tats for reducers, remove commented codeaa and changed addUserReducer to userReducer

ayodelevm committed:
Merge pull request #48 from ayodelevm/chore/tests-client-side

#150790281 client side tests

ayodelevm committed:
feat(test/implement-feedback): Implement feedback for tests and features. Details are:
- Refactor Js Doc
- Separate authorize routes into different files
- Complete naming of event in functions
- Change the naming of some files, functions and response object
- Split authorize routes file
- Update ui styling
- Adding to test-cases

ayodelevm committed:
Update ui styling

ayodelevm committed:
Create LICENSE.MD

ayodelevm committed:
Merge pull request #49 from ayodelevm/add-license-1

Create LICENSE.MD

ayodelevm committed:
Add FAQ, License, Limitations and how to contribute to the project

ayodelevm committed:
Merge pull request #50 from ayodelevm/chore/tests-feedback-implementation

Chore/tests feedback implementation

ayodelevm committed:
feat(test/implement-feedback): Implement feedback for tests and features. Details are:
- implement pagination on server side
- add more tests
- refactor controllers
- change some namings

ayodelevm committed:
converting index.jsx to pascal case

ayodelevm committed:
chore/tests-feedback-implementation

ayodelevm committed:
converting index.jsx to pascal case

ayodelevm committed:
chore(feedback-implementation): implement feedback for server side and client side as detailed below:
- [x] Change error message When the field is empty on the to add users to a group
- [x] Solve for situations when your API is consumed and instead of an integer being supplied as an id a string is supplied.
- [x] On GitHub repo's wiki, moving the border beneath the answer to a relevant question for the FAQs.
- [x] Add sample env file to better explain where and what parameter should be added
- [x] Move image at  https://www.conncoll.edu? to a personal source and reference from there
- [x] Move test mock data to a separate file for both your server side and client side test.
- [x] Make test files match in naming convention to the original file
- [x] Add JSDOCS methods and functions with JSDOcs and resolve on your server side and client side
- [x] Change all jsx files to Pascal case
- [x] Changed the use of the parameter ‘data’
- [x] Ensure that filenames should match default exports
- [x] review status codes used especially for when the server understands the syntax
- [x] Change success message for user profile picture uploads
- [x] Remove command to disable aslant inside files
- [x] Remove console.log in your code.
- [x] Move some logic from reducer to action
- [x] Change the direction of scroll for priority select box from down to up
- [x] Change placeholder for the reset password form
- [x] Add default value for priority select box and remove ability to search
- [x] Remove browser console error
- [x] Remove all commented out code before your push to your repo
- [x] Enforce Max Length rule, set to 80 characters
- [x] Write more tests for server and client side
- [x] Improve ui design

ayodelevm committed:
chore(prepare for deployment): changes includes:
- [x] Use webpack to extract css into separate file
- [x] Take out the Installation and Setup sections, replace the logo at the top left in api documentation
- [x] Setup web pack for production build
- [x] Serve api documentation from within the app itself
- [x] Add loader to forgot password

ayodelevm committed:
Merge pull request #51 from ayodelevm/chore/tests-feedback-implementation

Chore/tests feedback implementation

ayodelevm committed:
Merge pull request #52 from ayodelevm/chore/deployment-branch

chore/feedback/deployment

ayodelevm committed:
Merge pull request #53 from ayodelevm/chore/feedback-implementation

Chore/feedback implementation

ayodelevm committed:
Merge pull request #54 from ayodelevm/chore/tests-feedback-implementation

Chore/tests feedback implementation

ayodelevm committed:
chore(feedback implentation): applied max-len rule in my test files

ayodelevm committed:
Merge pull request #55 from ayodelevm/chore/tests-feedback-implementation

chore/ feedback implementation

ayodelevm committed:
Uncomment statement in file

ayodelevm committed:
chore(responsive-ui): changes include:
- improve responsiveness in ui
- handle more edge cases in reset password

ayodelevm committed:
Merge pull request #56 from ayodelevm/chore/responsive-ui

chore(responsive-ui): changes include:
